### DISCUSSION TITLE
Image cache in the browser memory

### DISCUSSION THREAD

**Discussion by 2020-byte** [supabase/supabase#28993] [Questions] [answered]
Is there a way to cache images fetched from a private bucket in Supabase in the browser memory when using Next.js so that the next time the image URL is used, the status is 200 from cache memory?

Is a signed URL also using a CDN? It takes about 1 to 3 seconds to load an image, which seems too slow.

**Comment by GaryAustin1** [comment, marked as answer]
Signed URL's are stored in the CDN just like public images, but only for that signed URL.  If you get a new signed URL for the file it would not be stored in the CDN.   Browsers should also cache both.

Supabase is not involved in caching locally other than what the browser may do.

**Reply by 2020-byte** [reply]
When loading images from a CDN, does it return a 200 from cache memory? In my case, I've always received only a 200 status when using the same signed URL, and takes more than 3 seconds to fetch a webp-type image.

<img width="783" alt="image" src="https://github.com/user-attachments/assets/a6eb88cf-d725-4496-ade1-af292f1d19d7">

**Reply by 2020-byte** [reply]
I was being foolish. I was generating a new signed URL every time I refreshed the page to load the image.
